hibernate-core/src/main/java/org/hibernate/engine/jdbc/spi/SqlStatementLogger.java

```diff
@@ -105,4 +119,30 @@ public void logStatement(String statement, Formatter formatter) {
 			System.out.println( "Hibernate: " + statement );
 		}
 	}
+
+	/**
+	 * Log a slow SQL query
+	 *
+	 * @param sql The SQL query.
+	 * @param startTime Start time in milliseconds.
+	 */
+	@AllowSysOut
+	public void logSlowQuery(String sql, long startTime) {
+		if ( logSlowQuery < 1 ) {
+			return;
+		}
+		assert startTime > 0 : "startTime is invalid!";
+
+		long spent = System.currentTimeMillis() - startTime;
+
+		assert spent >= 0 : "startTime is invalid!";
+
+		if ( spent > logSlowQuery ) {
+			String logData = "SlowQuery: " + spent + " milliseconds. " + sql;
```

Comment:
should there be something after milliseconds to clarify that what follows is the sql? Should the SQL be quoted?

Reply:
Fixed in 35a2d98abd80e3ed64734e22f69ec6277b08726a